File: four_pillars_of_oop/no_01_encapsulation/no_01_control/python/bank_account.py
class BankAccount:
    def __init__(self, initial_balance):
        self.__balance = initial_balance  # Private attribute

    def deposit(self, amount):
        if amount > 0:
            self.__balance += amount
        else:
            print("Deposit amount must be positive.")

    def withdraw(self, amount):
        if 0 < amount <= self.__balance:
            self.__balance -= amount
        else:
            print("Invalid withdraw amount.")

    def get_balance(self):
        return self.__balance

File: four_pillars_of_oop/no_01_encapsulation/no_01_control/java/BankAccount.java
package four_pillars_of_oop.no_01_encapsulation.no_01_control.java;

public class BankAccount {
    private double balance;  // Private attribute

    public BankAccount(double initialBalance) {
        if (initialBalance > 0) {
            this.balance = initialBalance;
        }
    }

    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
        } else {
            System.out.println("Deposit amount must be positive.");
        }
    }

    public void withdraw(double amount) {
        if (amount > 0 && amount <= balance) {
            balance -= amount;
        } else {
            System.out.println("Invalid withdraw amount.");
        }
    }

    public double getBalance() {
        return balance;
    }
}
